cache/network/download.go

```diff
@@ -66,5 +65,11 @@ func downloadFile(ctx context.Context, client *http.Client, url string, dest str
 	downloader := got.New()
 	downloader.Client = client
 
-	return downloader.Do(got.NewDownload(ctx, url, dest))
+	gDownload := got.NewDownload(ctx, url, dest)
```

Comment:
I understand this is because the client gets [set](https://github.com/melbahja/got/blob/master/download.go#L391) to the default client. We could use `downloader.Download()` which [uses](https://github.com/melbahja/got/blob/master/got.go#L36) the client set on it instead of setting it manually

Reply:
Correct. That said, I still feel like setting it on both is more future proof. The API of this 3rd party lib isn't the most bulletproof, that's why we missed this in the first place, and not sure if/how it'll change in the future. Maybe I'm too paranoid but this feels safer / more futureproof.

Reply:
There's also a difference between `downloader.Download()` and `got.NewDownload(ctx, url, dest)` -> `.Download()` does not have a `ctx` parameter.

Reply:
That said I just found that there's a `NewWithContext` which can set the context on the downloader.